engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/sink/TableSinkFactory.scala

```diff
@@ -34,38 +34,15 @@ import pl.touk.nussknacker.engine.util.sinkvalue.SinkValue
 import scala.collection.immutable.ListMap
 import scala.jdk.CollectionConverters._
 
-object TableSinkFactory {
-  private val valueParameterName: ParameterName = ParameterName("Value")
-  private val rawValueParameterDeclaration =
-    ParameterDeclaration.lazyMandatory[AnyRef](valueParameterName).withCreator()
-
-  private val tableNameParameterName              = TableSourceFactory.tableNameParamName
-  private val rawModeParameterName: ParameterName = ParameterName("Raw editor")
-
-  private val rawModeParameterDeclaration = ParameterDeclaration
-    .mandatory[Boolean](rawModeParameterName)
-    .withCreator(c => c.copy(defaultValue = Some(Expression.spel("false")), editor = Some(BoolParameterEditor)))
-
-  private val restrictedParamNamesForNonRawMode: Set[ParameterName] = Set(
-    tableNameParameterName,
-    rawModeParameterName
-  )
-
-}
-
-final case class TransformationState(table: TableDefinition, valueParam: SchemaBasedParameter)
-
 class TableSinkFactory(sqlStatements: List[SqlStatement])
     extends SingleInputDynamicComponent[Sink]
     with SinkFactory
     with BoundedStreamComponent {
 
   @transient
-  private lazy val tableDefinitions = TablesExtractor.extractTablesFromFlinkRuntimeUnsafe(sqlStatements)
-
-  override type State = TransformationState
+  private lazy val tablesDiscovery = TablesDefinitionDiscovery.prepareDiscoveryUnsafe(sqlStatements)
```

Comment:
How about moving this to ComponentProvider level to have a single TablesDefinitionDiscovery for sink and source?

Reply:
It is impossible because `TablesDefinitionDiscovery` is not serializable - it holds fat `TableEnvironmnet` object